c10/util/complex_type.h

```diff
@@ -119,110 +119,116 @@ namespace c10 {
 //  - thrust::complex only support float and double
 
 template<typename T>
-struct complex;
-
-template<typename T>
-struct alignas(sizeof(T) * 2) complex_common {
+struct alignas(sizeof(T) * 2) complex {
   using value_type = T;
 
-  T storage[2] = {T(), T()};
+  T real_ = T(0);
```

Comment:
nit: can we name them `real` and `imag` to keep it similar to the (TBA) python API

Reply:
No, there is already a function named `real`